homeassistant/helpers/config_validation.py

```diff
@@ -385,8 +387,47 @@ def validator(value):
     return validator
 
 
+def list_or_comma_separated(member_validator=match_all) -> Callable:
```

Comment:
I like the composition approach by @kellerza.

As this is a special case implementing a pattern that we do not want to promote, I would suggest we move this into the customize file. We do not want to promote it because normally when people define a list we want them to use a YAML list.